### PR TITLE
TEET-991 Fixes for entity linking

### PR DESCRIPTION
**Pull request by aleksievelizhanin**
Implemented the following: 
AC1 Order lists

- tasks by translation name alphabetically
- estates by number value
- land units by name alphabetically
- files by description alphabetically

AC2 Add confirmatiion modal dialog to delete link button. Show successful action confirmation in snack-bar after user has confirmed the action.

AC3 Increase width of the link entity selector.

AC4 links to land page in new tab/window with selected estate/unit and its estate/unit opened and scrolled into view

### CODE REVIEW

File: app/backend/src/clj/teet/file/file_db.clj [hunk 1]
```diff
@@ -193,8 +193,8 @@
         next-version (fn [file]
                        (some #(when (= (:db/id file)
                                        (get-in % [:file/previous-version :db/id])) %)
-                             next-versions))]
-    (vec
+                             next-versions))
+    meta-files (vec
```
Comment (tatut) [inline app/backend/src/clj/teet/file/file_db.clj:197]: fix indentation

Comment (tatut) [reply app/backend/src/clj/teet/file/file_db.clj:197]: the name `meta-files` doesn't seem right... perhaps it should be moved out of let and the sorting just wrapped around the `for`

File: app/backend/src/clj/teet/link/link_queries.clj [hunk 2]
```diff
@@ -15,35 +15,38 @@
...
-    (let [all-project-tasks
-          (d/q '[:find (pull ?t [:task/type :db/id])
-                 :where
-                 [?p :thk.project/lifecycles ?l]
-                 [?l :thk.lifecycle/activities ?a]
-                 [?a :activity/tasks ?t]
-                 [(missing? $ ?t :meta/deleted?)]
-                 :in $ ?p]
-               db project)
+                 (let [all-project-tasks
```
Comment (tatut) [inline app/backend/src/clj/teet/link/link_queries.clj:18]: Fix indentation, `with-` macros indent at 2

File: app/backend/src/clj/teet/link/link_queries.clj [hunk 3]
```diff
@@ -58,9 +61,10 @@
          (map (fn [[key properties]]
                 (assoc properties :link/external-id (name key))))
          (filterv #(or (string/contains-words? (:TUNNUS %) text)
-                       (string/contains-words? (:L_AADRESS %) text))))))
+                       (string/contains-words? (:L_AADRESS %) text)))
+         (sort-by :L_AADRESS))))
 
-(defn search-estate [db {:keys [api-url api-secret]} project text]
+  (defn search-estate [db {:keys [api-url api-secret]} project text]
```
Comment (tatut) [inline app/backend/src/clj/teet/link/link_queries.clj:67]: extra space at beginning?

File: app/backend/src/clj/teet/link/link_queries.clj [hunk 4]
```diff
@@ -74,7 +78,9 @@
          distinct
          (filterv #(and (not (str/blank? (:KINNISTU %)))
                         (string/contains-words? (:KINNISTU %) text)))
-         (map #(assoc % :link/external-id (:KINNISTU %))))))
+         (map #(assoc % :link/external-id (:KINNISTU %)))
+         (sort-by #(Integer/parseInt (:link/external-id %)))
+         )))
```
Comment (tatut) [inline app/backend/src/clj/teet/link/link_queries.clj:83]: ending parens on same line

File: app/frontend/src/cljs/teet/project/land_controller.cljs [hunk 5]
```diff
@@ -44,6 +45,11 @@
                                   (not (:selected? unit)))))
     cad-units))
 
+(defn cadastral-unit-dom-id [id]
```
Comment (tatut) [inline app/frontend/src/cljs/teet/project/land_controller.cljs:48]: these 2 should probably be private (`defn-`)

File: app/frontend/src/cljs/teet/project/land_controller.cljs [hunk 6]
```diff
@@ -353,10 +359,20 @@
...
+                         (cadastral-unit-dom-id (get-in app [:query :unit-id])))) 1000) result)))
             (when linked-estate
               (fn [e!]
-                (e! (->ToggleOpenEstate linked-estate)))))))
+                (let [result (e! (->ToggleOpenEstate linked-estate))]
+                  (js/setTimeout
+                    #(animate/scroll-into-view!
+                       (js/document.getElementById
+                         (estate-dom-id linked-estate))) 1000)
+                  result
```
Comment (tatut) [inline app/frontend/src/cljs/teet/project/land_controller.cljs:374]: ending parens

**Review by tatut — COMMENTED** [review]
some style issues

File: app/backend/src/clj/teet/file/file_db.clj (hunk 1 above)
Comment (aleksievelizhanin) [reply app/backend/src/clj/teet/file/file_db.clj:197]: It is not clear what is the defect here

Comment (tatut) [reply app/backend/src/clj/teet/file/file_db.clj:197]: indentation, indent meta-files starting from the same column as other let bindings

the other is a comment about the name, which isn't clear to me... why are they "meta" files?

File: app/frontend/src/cljs/teet/project/land_controller.cljs (hunk 5 above)
Comment (aleksievelizhanin) [reply app/frontend/src/cljs/teet/project/land_controller.cljs:48]: There might be a problem with warning like ".. var: teet.project.land-controller/estate-dom-id is not public .." then those functions called from land_view.cljs

Comment (tatut) [reply app/frontend/src/cljs/teet/project/land_controller.cljs:48]: My bad, I thought these were new and only used here.

**Pull request merged**